packages/expo-sqlite/src/Storage.ts

```diff
@@ -412,6 +413,39 @@ export class SQLiteStorage {
   //#endregion
 }
 
+//#region Web Storage compatible API
+
+class WebStorageWrapper {
+  constructor(private readonly storage: SQLiteStorage) {}
+
+  clear() {
+    this.storage.clearSync();
+  }
+
+  getItem(key: string): string | null {
+    return this.storage.getItemSync(key);
+  }
+
+  key(index: number): string | null {
+    const keys = this.storage.getAllKeysSync();
+    return keys[index] ?? null;
+  }
+
+  removeItem(key: string) {
+    this.storage.removeItemSync(key);
+  }
+
+  setItem(key: string, value: string) {
+    this.storage.setItemSync(key, value);
+  }
+
+  get length(): number {
+    return this.storage.getAllKeysSync().length;
```

Comment:
Similar comment about performance and an area for improvement.

Reply:
since we have sqlite under the hood, we can do some optimization. improved in https://github.com/expo/expo/pull/38699/commits/a43c65347f9192ae706e582c3abd8f4fdfa6361f